mods/flowers/init.lua

```diff
@@ -235,15 +235,22 @@ minetest.register_node("flowers:waterlily", {
 		fixed = {-0.5, -0.5, -0.5, 0.5, -0.4375, 0.5}
 	},
 
-	on_place = function(itemstack, _, pointed_thing)
+	on_place = function(itemstack, placer, pointed_thing)
 		local pos = pointed_thing.above
 		local node = minetest.get_node(pointed_thing.under).name
 		local def = minetest.registered_nodes[node]
 		if def and def.liquidtype == "source" and minetest.get_item_group(node, "water") > 0 then
 			minetest.set_node(pos, {name = "flowers:waterlily", param2 = math.random(0, 3)})
 			if not minetest.setting_getbool("creative_mode") then
-				itemstack:take_item()
-				return itemstack
+				if itemstack:get_count() == 1 then
+					local inv = placer:get_inventory()
+					minetest.after(0, function()
+						inv:set_stack("main", placer:get_wield_index(), nil)
```

Comment:
doing it this way is dangerous, if the player places it and then switches to another item, the wrong one gets removed